### PR TITLE
Added \$project stage

### PR DESCRIPTION
**Pull request by sourabhbagrecha**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -61,4 +63,15 @@ const applyGroupStage = (query: any, data: any) => {
   return results;
 };
 
+const applyProjectStage = (query: any, data: any) => {
+  return data.map((item: any) => {
+    const result: any = {};
+    for (let [key, value] of Object.entries(query)) {
+      if (value === 1) result[key] = item[key];
```
Comment (takameyer) [inline src/index.ts:70]: I was reading the documentation for `$project` and noticed that it's possible to set this to `true`, `0` or `false`.  And I think `_id` is included, unless explicitly removed (though I could be wrong)

**Review by takameyer — COMMENTED** [review]
Looks great.  Just a comment

File: src/index.ts (hunk 1 above)
Comment (sourabhbagrecha) [reply src/index.ts:70]: That's true. I just added the inclusions as a starting point, I am working on adding the exclusion logic and will push the same soon. Thanks.

File: src/index.ts [hunk 2]
```diff
@@ -61,4 +63,15 @@ const applyGroupStage = (query: any, data: any) => {
...
+      else if (typeof value === "string") result[key] = item[value.slice(1)];
```
[the remark this answers is not in the document]
Comment (sourabhbagrecha) [reply src/index.ts:71]: I agree. On it!

**Review by takameyer — APPROVED** [review]

**Pull request merged**